pkg/storageincentives/redistribution/redistribution.go

```diff
@@ -45,27 +45,27 @@ func New(
 	logger log.Logger,
 	txService transaction.Service,
 	incentivesContractAddress common.Address,
-) *Service {
+) Contract {
```

Comment:
although this is fine here, but if it's not necessary then generally it's recommended to lean towards returning concrete types